src/analysis/normed_space/hahn_banach.lean

```diff
@@ -21,6 +21,32 @@ Prove more corollaries
 
 -/
 
+universes u v
+
+/-- Hahn-Banach theorem for continuous linear functions. -/
+class has_exists_extension_norm_eq (𝕜 : Type v)
+  [nondiscrete_normed_field 𝕜] [normed_algebra ℝ 𝕜] : Prop :=
+(exists_extension_norm_eq :
+  ∀ (E : Type u)
+  [normed_group E] [normed_space 𝕜 E]
+  (p : subspace 𝕜 E)
+  (f : p →L[𝕜] 𝕜),
+  ∃ g : E →L[𝕜] 𝕜, (∀ x : p, g x = f x) ∧ ∥g∥ = ∥f∥)
+
+/-- The norm of `x` as an element of 𝕜 -/
```

Comment:
Could you expand a little bit the docstring, mentioning in particular that you assume that `𝕜` is an algebra over the reals, and saying how this can be useful?